### PR TITLE
[DAT-484] - Add edition view in the payment method section

### PR DESCRIPTION
**Pull request by jhoffman-ms**
Add the edition view in the payment method section when the payment type is 'Credit Card'

Screenshot: 

image

**Task:** [DAT-484](https://makingsense.atlassian.net/jira/software/projects/DAT/boards/62?selectedIssue=DAT-484)

### CODE REVIEW

File: src/components/ChangePlan/Checkout/PaymentMethod/PaymentMethod.js [hunk 1]
```diff
@@ -35,33 +40,138 @@ const paymentMethods = [
...
+            return <CreditCard optionView={optionView}></CreditCard>;
+          default:
+            return null;
+        }
+      })()}
+    </>
+  );
+};
+
+const PaymentNotes = ({ paymentMethodType }) => {
```
Comment (jcamposmk) [inline src/components/ChangePlan/Checkout/PaymentMethod/PaymentMethod.js:122]:
```suggestion
const PaymentNotes = ({ paymentMethodType }) => {
  const intl = useIntl();
  const _ = (id, values) => intl.formatMessage({ id: id }, values);

  switch (paymentMethodType) {
    case paymentType.creditCard:
      return (
        <FieldGroup>
          <li className="field-item">
            <div className="dp-wrap-subscription">
              <label>{_('checkoutProcessForm.payment_method.considerations')}</label>
              <p style={considerationNoteStyle}>
                <FormatMessageWithBoldWords id="checkoutProcessForm.payment_method.considerations_credit_card_note_1" />
              </p>
              <br />
              <p style={considerationNoteStyle}>
                <FormatMessageWithBoldWords id="checkoutProcessForm.payment_method.considerations_credit_card_note_2" />
              </p>
            </div>
          </li>
        </FieldGroup>
      );
    default:
      return null;
  }
};
```

File: src/components/ChangePlan/Checkout/Discounts/Discounts.test.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+      sessionPlan={sessionPlan}
+      selectedPlanDiscount={selectedPlanDiscount}
+      disabled={disabled}
+      handleChange={mockedhandleChange}
+    />
+  </IntlProvider>
+);
+
+describe('Discount component', () => {
+  afterEach(cleanup);
```
Comment (jcamposmk) [inline src/components/ChangePlan/Checkout/Discounts/Discounts.test.js:22]: It's not necessary. [More details](https://github.com/FromDoppler/doppler-webapp/pull/1504)

File: src/components/ChangePlan/Checkout/Discounts/Discounts.test.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+  it('should show discount message when subscription is quarterly or higher', async () => {
+    // Arrange
+    const fakeSessionPlan = {
+      idPlan: 42,
+      planSubscription: 6,
+    };
+
+    // Act
+    const { container } = render(
```
Comment (jcamposmk) [inline src/components/ChangePlan/Checkout/Discounts/Discounts.test.js:32]: Why `container` ? there is another easier way . [See Guide Lines for more information](https://github.com/FromDoppler/doppler-webapp#test)

**Comment by jcamposmk** [comment]
**An act warning:**
image

**there are many unused imports and parameters. Can you remove it?**
image

File: src/components/ChangePlan/Checkout/Discounts/Discounts.test.js (hunk 2 above)
Comment (jhoffman-ms) [reply src/components/ChangePlan/Checkout/Discounts/Discounts.test.js:22]: 👍

File: src/components/ChangePlan/Checkout/Discounts/Discounts.test.js (hunk 3 above)
Comment (jhoffman-ms) [reply src/components/ChangePlan/Checkout/Discounts/Discounts.test.js:32]: I used ```container``` because I needed find an element by class.

Comment (jhoffman-ms) [reply src/components/ChangePlan/Checkout/Discounts/Discounts.test.js:32]: for example: ```container.querySelector('.dp-wrap-subscription')```. Do you know another way to find it?

File: src/components/ChangePlan/Checkout/PaymentMethod/PaymentMethod.js (hunk 1 above)
Comment (jhoffman-ms) [reply src/components/ChangePlan/Checkout/PaymentMethod/PaymentMethod.js:122]: 👍

**Comment by jhoffman-ms** [comment]
@jcamposmk, the last two point were resolved.

**Comment by jcamposmk** [comment]
Good job! :+1:

**Pull request merged**